Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.cpp

```diff
@@ -78,17 +139,17 @@ namespace ShaderManagementConsole
             m_shaderOptionCount, m_documentId, &ShaderManagementConsoleDocumentRequestBus::Events::GetShaderOptionDescriptorCount);
 
         // Only clear the table if the number of columns or rows have changed
-        if (rowCount() != m_shaderVariantCount || columnCount() != m_shaderOptionCount)
+        if (rowCount() != m_shaderVariantCount || (columnCount() + 1) != m_shaderOptionCount)
```

Comment:
Adding the deleter widgets column created the need for this index hack.
I personally feel it as risky and ugly, so if anybody sees a method to virtualize column access to hide that index hack under a safer accessor that'd be great.